File: android/src/main/java/com/mynitromodule/HybridMyNitroModule.kt
package com.mynitromodule

import android.os.Build
import com.margelo.nitro.mynitromodule.HybridMyNitroModuleSpec
import com.margelo.nitro.mynitromodule.DeviceInfo
import com.margelo.nitro.core.Promise
// Coroutines imports removed for simplified Promise implementation

class HybridMyNitroModule: HybridMyNitroModuleSpec() {    
    override fun sum(num1: Double, num2: Double): Double {
        return num1 + num2
    }
    
    override fun multiply(num1: Double, num2: Double): Double {
        return num1 * num2
    }
    
    override fun greet(name: String): String {
        return "Hello, $name! Welcome to Nitro modules."
    }
    
    override fun reverseString(input: String): String {
        return input.reversed()
    }
    
    override fun getArraySum(numbers: DoubleArray): Double {
        return numbers.sum()
    }
    
    override fun getDeviceInfo(): DeviceInfo {
        return DeviceInfo(
            platform = "Android",
            model = Build.MODEL,
            version = Build.VERSION.RELEASE
        )
    }
    
    override fun delayedGreeting(name: String, delayMs: Double): Promise<String> {
        // For now, return a resolved promise with a simple message
        // TODO: Implement proper async delay once we understand Nitro Promise API better
        return Promise.resolved("Hello, $name! This greeting was delayed by ${delayMs.toInt()}ms (simulated).")
    }
    
    override fun isEven(number: Double): Boolean {
        return number.toInt() % 2 == 0
    }
}

File: ios/HybridMyNitroModule.swift
//
//  HybridMyNitroModule.swift
//  Pods
//
//  Created by  on 8/9/2025.
//

import Foundation
import UIKit

class HybridMyNitroModule: HybridMyNitroModuleSpec {
    func sum(num1: Double, num2: Double) throws -> Double {
        return num1 + num2
    }
    
    func multiply(num1: Double, num2: Double) throws -> Double {
        return num1 * num2
    }
    
    func greet(name: String) throws -> String {
        return "Hello, \(name)! Welcome to Nitro modules."
    }
    
    func reverseString(input: String) throws -> String {
        return String(input.reversed())
    }
    
    func getArraySum(numbers: [Double]) throws -> Double {
        return numbers.reduce(0, +)
    }
    
    func getDeviceInfo() throws -> [String: Any] {
        return [
            "platform": "iOS",
            "model": UIDevice.current.model,
            "version": UIDevice.current.systemVersion
        ]
    }
    
    func delayedGreeting(name: String, delayMs: Double) -> Promise<String> {
        return Promise.async { resolve in
            DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(Int(delayMs))) {
                resolve("Hello, \(name)! This greeting was delayed by \(Int(delayMs))ms.")
            }
        }
    }
    
    func isEven(number: Double) throws -> Bool {
        return Int(number) % 2 == 0
    }
}
